### PR TITLE
Change fieldtype from float to currency

### PR DESCRIPTION
**Pull request by rohitwaghchaure**

### CODE REVIEW

File: erpnext/selling/doctype/sales_order_item/sales_order_item.json [hunk 1]
```diff
@@ -563,7 +563,7 @@
    "columns": 0, 
    "depends_on": "eval:doc.margin_type && doc.price_list_rate", 
    "fieldname": "total_margin", 
-   "fieldtype": "Float", 
+   "fieldtype": "Currency",
```
Comment (nabinhait) [inline erpnext/selling/doctype/sales_order_item/sales_order_item.json]: @rohitwaghchaure You have to change fieldtype for margin in all other doctypes as well.

Comment (rohitwaghchaure) [inline erpnext/selling/doctype/sales_order_item/sales_order_item.json]: Done

**Comment by nabinhait** [comment]
@rohitwaghchaure why you did not change fieldtype for "Margin Rate or Amount" field? Also same in Pricing Rule?

**Comment by rohitwaghchaure** [comment]
@nabinhait 
"Margin Rate or Amount" field is used as a percentage / amount. So user may get confused if fieldtype is currency and value is percentage

**Pull request merged**